e2etest/pkg/executor/executor.go

```diff
@@ -67,8 +69,18 @@ func (p *podExecutor) Exec(cmd string, args ...string) (string, error) {
 		return "", errors.New("the kubectl parameter is not set")
 	}
 	fullargs := append([]string{"exec", p.name, "-n", p.namespace, "-c", p.container, "--", cmd}, args...)
-	out, err := exec.Command(Kubectl, fullargs...).CombinedOutput()
-	return string(out), err
+
+	c := exec.Command(Kubectl, fullargs...)
+	var stdoutBuf, stderrBuf bytes.Buffer
+	c.Stdout = &stdoutBuf
+	c.Stderr = &stderrBuf
+	if err := c.Run(); err != nil {
+		return "", fmt.Errorf("kubectl exec failed: %w, stderr: %s", err, stderrBuf.String())
+	}
+	if stderrBuf.String() != "" {
+		return "", fmt.Errorf("kubectl exec succeeded but stderr: %s", stderrBuf.String())
```

Comment:
Do you think if the err is nil, but the stderr is not empty, should we return an error?

In our flake  the `kubectl exec` is not failing is just adding some stderr, so the final return string
is not a valid json when parsed from the next function (that parses json into route struct)

```
Failed to parse routes E0910 18:08:54.853077  339136 websocket.go:296] Unknown stream id 1, discarding message\n

{\n\"default\":{\n \"vrfId\": 0,\n \"vrfName\": \"default\",\n \"tableVersion\": 0,\n \"routerId\": \"172.19.0.2\",\n \"defaultLocPrf\": 100,\n \"localAS\": 64512,\n \"routes\": {  }  } \n,\n\"red\":{\n \"vrfId\": 5,\n \"vrfName\": \"red\",\n \"tableVersion\": 0,\n \"routerId\": \"172.31.0.4\",\n \"defaultLocPrf\": 100,\n \"localAS\": 64513,\n \"routes\": {  }  } \n}\n
````

For that flake I am not sure a retry logic makes sense given the error is nil

Reply:
If that's the case, then I think the right thing to do is to return the sdterr only after a proper failure, and the std output otherwise.